Update arrays.md (#6088)
ballerina-platform/ballerina-distribution: examples/arrays/arrays.md

An array can be used to hold a set of values of the same type. The array type can be defined as `T[n]` in which `T` is the element type and `n` is the length of the array. `n` must be an integer literal or constant reference of type `int`. Optionally, you can create a variable-length array by defining an array without `n` as `T[]`.

The length of the array can be inferred from the context by defining the array as `T[*]`. The length of the array should be known in compile time.

> An array of inferred length is also a fixed-length array, where the length is inferred from the right-hand side during initialization and cannot be changed afterward.

Reviewer: ```suggestion
An array with an inferred length is also a fixed-length array, where the length, inferred from the context, cannot be changed afterward.
```

The length of the array can be inferred from the context by defining the array as `T[*]`. The length of the array should be known in compile time.

# Quick Comparison of Array Types in Ballerina

| **Array Type**             | **Declaration Syntax** | **Example Syntax**                  | **Length Known at Compile Time** | **Resizable** |
|----------------------|------------------------|-------------------------------------|----------------------------------|---------------|
| Fixed-length   | `T[n] <arrayName>;`               | `int[3] nums = [1, 2, 3];`          | Yes ✅                           | No ❌         |
| Variable-length      | `T[] <arrayName>;`                | `int[] even = [2, 4];`           | No ❌                            | Yes ✅        |

> | Inferred-length      | `T[*] <arrayName>;`            | `string[*] colors = ["red"];`       | Yes ✅                           | No ❌         |

Reviewer: Since inferred-length arrays are also fixed-length arrays, I'm not sure this classification works. And if it's just fixed vs variable length, I think the descriptions work without the table too.

> 📌 **Note**:
> - `T` Represents any valid data type (e.g., `int`, `string`, `float`, etc.)
> - `n` Specifies the exact number of elements required in the array. 
> - Fixed-length and inferred-length arrays cannot change their size once initialized.  

> > - Variable-length arrays (also known as open arrays) allow dynamic resizing using methods like `.push()` and `.remove()`.

Reviewer: @Adeegithub, thank you for your contribution!

We generally try to leave the actual code examples for the code file itself, to avoid repetition and also for readability. Given that most of these points are already covered in the previous paragraphs, shall we just add the missing part (array of inferred length also being fixed-length) as a sentence instead?

Author: Thanks for the suggestion! That makes sense—I was aiming to provide a quick reference summary, especially for readers who might skim through. But I agree that avoiding repetition and keeping the content clean is important.

I’m happy to revise it and add the missing clarification about inferred-length arrays being fixed-length directly in the paragraph instead. Appreciate the feedback!
